partner_delivery_zone/tests/test_partner_delivery_zone.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo.tests import SavepointCase
+
+
+class TestPartnerDeliveryZone(SavepointCase):
+    at_install = False
+    post_install = True
```

Comment:
@sergio-teruel @pedrobaeza

Reply:
Well, we put this in our modules by default, and prefer to leave it, as we are maintaining them.

Reply:
@pedrobaeza To make reviewers work easy, I will port the debate at higher level. I want to avoid possible future debates (time eating) if some modules collide on tests execution (you are not maintainer on all modules).

I will stop merging such ones before decision.

Moving to https://github.com/OCA/odoo-community.org/

Reply:
@rousseldenis OK, we are going to remove this part, but as you say, take into account that we are not going to face this debate again. As soon as we have an error in our integration tests because of this, I'm going to put it directly. @sergio-teruel please remove it meanwhile.

Reply:
> I'm going to put it directly
Depending on future guidelines uh ? ;-)

Reply:
You can't enforce that rule, and if so, we will be out of OCA automatically. If you don't perform integration tests nor continuous deployment, it doesn't mean that others shouldn't do them.

Reply:
> If you don't perform integration tests nor continuous deployment

Of course we do (that's the main part of process)! But you can't do it for a whole OCA repository ! Moreover, integration test cases could be unique by project!

I think you don't understand difference between integration tests and unit tests.

> Well, we put this in our modules by default

I think that's quite new in your dev processes as it should have been seen before. Along all versions of Odoo, the tests process hasn't changed a lot, but always been at install. So, we were wrong for so many years ?

Reply:
> Of course we do (that's the main part of process)! But you can't do it for a whole OCA repository ! Moreover, integration test cases could be unique by project!
> I think you don't understand difference between integration tests and unit tests.

That's the key: we detect problems when we perform integration tests on our projects, and this is when we add this post_install, for getting them green. As it's more work to detect it afterwards, now we put them by default. And it's curious that we haven't found the other problem yet... (well, in fact it's not curious, but a very specific case that rarely or never happens).

> I think that's quite new in your dev processes as it should have been seen before. Along all versions of Odoo, the tests process hasn't changed a lot, but always been at install. So, we were wrong for so many years ?

Not so new. We are performing this for about 6-7 months. There are already 30-35 modules with these keys across OCA.

Reply:
> Not so new. We are performing this for about 6-7 months

Quite comparing to years running travis

Reply:
Yeah, because Travis only performs unit tests + repository integration tests, not full integration tests.

Reply:
> Yeah, because Travis only performs unit tests + repository integration tests, not full integration tests.

So, you agree you introduce a new way of testing

Reply:
We are not debating about that. We are debating about forcing to not be possible to do this with OCA modules, what is what you are trying to do forbidding this by guidelines.

Reply:
IMHO you have to override that way of testing in your integration process. You cannot constrain developers to think about 'what could happen with ...possible... installed modules'.

Reply:
The same can apply to your reasoning. And this is not something that can be adapted easily. But we are not forcing you that. If we find any case that this is problematic, we solve it. You don't have to think about anything. And I repeat, till now, 0 problems in all your integration tests of our deployments in v9, v10 and v11, so the theoretical case where post_install is a problem hasn't arisen yet (and I doubt it does, at least because of the post_install part). Any way, we have already done a PR over Odoo for improving this situation and to not need post_install anymore, but this will be for v13 hopefully.